### PR TITLE
Calculate the projected volume correctly when picking up stacked items

### PR DESCRIPTION
**Pull request by sethsimon**
Fixes [#15728](https://github.com/CleverRaven/Cataclysm-DDA/issues/15728).

### CODE REVIEW

**Comment by BevapDin** [comment]
Looks way too complicated. The code in inventory_ui.cpp uses a much simpler logic: it creates a temporary player object as copy of the player character (`player tmp = g->u;`), removes the items to be dropped and calls `tmp.volume_carried()` to get the new volume.

**Comment by sethsimon** [comment]
Silly me, that idea never even crossed my mind. Thanks for the insight!

File: src/pickup.cpp [hunk 1]
```diff
@@ -753,16 +756,28 @@ void Pickup::pick_up( const tripoint &pos, int min )
...
                 if( getitem[idx] ) {
-                    if( getitem[idx].count != 0 && getitem[idx].count < here[idx].charges ) {
-                        item temp = here[idx];
-                        temp.charges = getitem[idx].count;
-                        new_weight -= temp.weight();
-                        new_volume -= temp.volume();
-                    } else {
-                        new_weight -= here[idx].weight();
-                        new_volume -= here[idx].volume();
+                    if(here[idx].count_by_charges()) {
```
Comment (kevingranade) [inline src/pickup.cpp]: You're shifting away from the project coding style here, parentheses are separated from their contents by a space.
Also on line 762-763, else goes on the dame line as its matching bracket.

**Comment by kevingranade** [comment]
This didn't quite fix it, it's still prompting to wield the item, but later in the process now.

**Comment by kevingranade** [comment]
Rebased a bit, which means github didn't recognize that I merged it.

**Pull request closed without merging**